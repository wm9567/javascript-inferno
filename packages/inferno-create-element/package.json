{
  "name": "inferno-create-element",
  "version": "6.0.0-rc.1",
  "license": "MIT",
  "description": "Provides methods to create Inferno VNodes",
  "author": {
    "name": "Dominic Gannaway",
    "email": "dg@domgan.com"
  },
  "bugs": {
    "url": "https://github.com/infernojs/inferno/issues"
  },
  "homepage": "https://github.com/infernojs/inferno#readme",
  "main": "index.js",
  "module": "dist/index.esm.js",
  "dev:module": "dist/index.dev.esm.js",
  "typings": "dist/index.d.ts",
  "keywords": [
    "babel",
    "react",
    "inferno",
    "framework",
    "interfaces",
    "user interfaces",
    "vdom"
  ],
  "dependencies": {
    "inferno": "^6.0.0-rc.1"
  },
  "devDependencies": {
    "inferno-shared": "^6.0.0-rc.1",
    "inferno-vnode-flags": "^6.0.0-rc.1"
  },
  "rollup": {
    "bundledDependencies": [
      "inferno-shared",
      "inferno-vnode-flags"
    ],
    "moduleName": "Inferno",
    "moduleGlobals": {
      "inferno": "Inferno"
    }
  }
}
