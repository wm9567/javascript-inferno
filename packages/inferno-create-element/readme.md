# inferno-create-element
> Inferno package with helpers to create Inferno VNodes

## Install

```
npm install inferno-component
```

## Usage

```
const createElement = InfernoCreateElement;

InfernoDOM.render(createElement('div', { className: 'test' }, 'I'm a child!'), document.body);
```



