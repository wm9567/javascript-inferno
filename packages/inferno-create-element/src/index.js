/**
 * Created by ADM on 09/04/2016.
 */
