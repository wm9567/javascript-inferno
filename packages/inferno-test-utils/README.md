# inferno-test-utils

> Suite of utilities for testing Inferno applications

## Install

```
npm install inferno --save
npm install inferno-test-utils --save-dev
```

## Contents

* [`renderIntoDocument`](#renderintodocumentvnodetree)
* [`findAllInRenderedTree`](#findallinrenderedtreerenderedtree-predicate)
* [`findAllInVNodeTree`](#findallinvnodetreevnodetree-predicate)
* [`scryRenderedDOMElementsWithClass`](#scryrendereddomelementswithclassrenderedtree-classname)
* [`findRenderedDOMElementWithClass`](#findrendereddomelementwithclassrenderedtree-classname)
* [`scryRenderedDOMElementsWithTag`](#scryrendereddomelementswithtagrenderedtree-tagname)
* [`findRenderedDOMElementWithTag`](#findrendereddomelementwithtagrenderedtree-tagname)
* [`scryRenderedVNodesWithType`](#scryrenderedvnodeswithtyperenderedtree-type)
* [`findRenderedVNodeWithType`](#findrenderedvnodewithtyperenderedtree-type)
* [`scryVNodesWithType`](#scryvnodeswithtypevnodetree-type)
* [`findVNodeWithType`](#findvnodewithtypevnodetree-type)
* [`isVNode`](#isvnodeinstance)
* [`isVNodeOfType`](#isvnodeoftypeinstance-type)
* [`isDOMVNode`](#isdomvnodeinstance)
* [`isDOMVNodeOfType`](#isdomvnodeoftypeinstance-type)
* [`isFunctionalVNode`](#isfunctionalvnodeinstance)
* [`isFunctionalVNodeOfType`](#isfunctionalvnodeoftypeinstance-type)
* [`isClassVNode`](#isclassvnodeinstance)
* [`isClassVNodeOfType`](#isclassvnodeoftypeinstance-type)
* [`isDOMElement`](#isdomelementinstance)
* [`isDOMElementOfType`](#isdomelementoftypeinstance-type)
* [`isRenderedClassComponent`](#isrenderedclasscomponentinstance)
* [`isRenderedClassComponentOfType`](#isrenderedclasscomponentoftypeinstance-type)

## Usage

### `renderIntoDocument(VNodeTree)`

Renders `VNodeTree` into a detached DOM element in the document and returns a rendered `VNode` tree.

_This function requires a DOM_.

```js
const VNodeTree = (
  <div className="outer">
    <SomeComponent className="inner"/>
  </div>
);
const renderedTree = renderIntoDocument(VNodeTree);
```

### `findAllInRenderedTree(renderedTree, predicate)`

Calls `predicate` with each `VNode` instance in `renderedTree`. 

Returns an array of `VNodes` where `predicate` returns `true`.

```js
const VNodeTree = (
  <div className="outer">
    <SomeComponent className="inner"/>
  </div>
);
const renderedTree = renderIntoDocument(VNodeTree);
const predicate = (VNode) => VNode.type === SomeComponent;
const result = findAllInRenderedTree(renderedTree, predicate);
```

### `findAllInVNodeTree(VNodeTree, predicate)`

Calls `predicate` with each `VNode` instance in `VNodeTree`. 

Returns an array of `VNodes` where `predicate` returns `true`.

```js
const VNodeTree = (
  <div className="outer">
    <SomeComponent className="inner"/>
  </div>
);
const predicate = (VNode) => VNode.type === SomeComponent;
const result = findAllInVNodeTree(VNodeTree, predicate);
```

### `scryRenderedDOMElementsWithClass(renderedTree, className)`

Returns an array of DOM elements with `className`.

`className` can be a space-separated string or an array of strings.

```js
const VNodeTree = (
  <div className="outer">
    <SomeComponent className="inner one"/>
    <SomeComponent className="inner two"/>
  </div>
);
const renderedTree = renderIntoDocument(VNodeTree);
const result1 = scryRenderedDOMElementsWithClass(renderedTree, 'inner');
const result2 = scryRenderedDOMElementsWithClass(renderedTree, 'inner one');
const result3 = scryRenderedDOMElementsWithClass(renderedTree, ['inner', 'two']);
const result4 = scryRenderedDOMElementsWithClass(renderedTree, 'three'); // Empty array
```

### `findRenderedDOMElementWithClass(renderedTree, className)`

Returns a single DOM element with `className`. If more than one matches are found, throws an error.

`className` can be a space-separated string or an array of strings.

```js
const VNodeTree = (
  <div className="outer">
    <SomeComponent className="inner one"/>
    <SomeComponent className="inner two"/>
  </div>
);
const renderedTree = renderIntoDocument(VNodeTree);
const result1 = scryRenderedDOMElementsWithClass(renderedTree, 'outer');
const result2 = scryRenderedDOMElementsWithClass(renderedTree, 'inner one');
// Will throw an error because more than 1 matches were found...
const result3 = scryRenderedDOMElementsWithClass(renderedTree, 'inner');
```

### `scryRenderedDOMElementsWithTag(renderedTree, tagName)`

Returns an array of DOM elements with `tagName`.

```js
const VNodeTree = (
  <div>
    <h1>Heading</h1>
    <p>Paragraph One</p>
    <p>Paragraph Two</p>
    <p>Paragraph Three</p>
  </div>
);
const renderedTree = renderIntoDocument(VNodeTree);
const result1 = scryRenderedDOMElementsWithTag(renderedTree, 'h1');
const result3 = scryRenderedDOMElementsWithTag(renderedTree, 'p');
const result4 = scryRenderedVNodesWithType(renderedTree, 'span'); // Empty array
```

### `findRenderedDOMElementWithTag(renderedTree, tagName)`

Returns a single DOM element with `tagName`. If more than one matches are found, throws an error.

```js
const VNodeTree = (
  <div>
    <h1>Heading</h1>
    <div>
      <p>Paragraph One</p>
      <p>Paragraph Two</p>
      <p>Paragraph Three</p>
    </div>
  </div>
);
const renderedTree = renderIntoDocument(VNodeTree);
const result1 = findRenderedDOMElementWithTag(renderedTree, 'h1');
// Will throw an error because more than 1 matches were found...
const result2 = findRenderedDOMElementWithTag(renderedTree, 'p');
```

### `scryRenderedVNodesWithType(renderedTree, type)`

Returns an array of rendered `VNodes` with `type`.

```js
const VNodeTree = (
  <div>
    <h1>Heading</h1>
    <SomeComponent/>
    <SomeComponent/>
  </div>
);
const renderedTree = renderIntoDocument(VNodeTree);
const result1 = scryRenderedVNodesWithType(renderedTree, 'h1');
const result2 = scryRenderedVNodesWithType(renderedTree, SomeComponent);
const result3 = scryRenderedVNodesWithType(renderedTree, 'p'); // Empty array
```

### `findRenderedVNodeWithType(renderedTree, type)`

Returns a single rendered `VNode` with `type`. If more than one matches are found, throws an error.

```js
const VNodeTree = (
  <div>
    <h1>Heading</h1>
    <p>Paragraph 1</p>
    <p>Paragraph 2</p>
    <SomeComponent/>
    <AnotherComponent/>
    <AnotherComponent/>
  </div>
);
const renderedTree = renderIntoDocument(VNodeTree);
const result1 = findRenderedVNodeWithType(renderedTree, 'h1');
const result2 = findRenderedVNodeWithType(renderedTree, SomeComponent);
// Will throw an error because more than 1 matches were found...
const result3 = findRenderedVNodeWithType(renderedTree, 'p');
const result4 = findRenderedVNodeWithType(renderedTree, AnotherComponent);
```

### `scryVNodesWithType(VNodeTree, type)`

Returns an array of `VNodes` with `type`.

```js
const VNodeTree = (
  <div>
    <h1>Heading</h1>
    <SomeComponent/>
    <SomeComponent/>
  </div>
);
const result1 = scryVNodesWithType(VNodeTree, 'h1');
const result2 = scryVNodesWithType(VNodeTree, SomeComponent);
const result3 = scryVNodesWithType(VNodeTree, 'p'); // Empty array
```

### `findVNodeWithType(VNodeTree, type)`

Returns a single `VNode` with `type`. If more than one matches are found, throws an error.

```js
const VNodeTree = (
  <div>
    <h1>Heading</h1>
    <SomeComponent/>
    <SomeComponent/>
  </div>
);
const result1 = findVNodeWithType(VNodeTree, 'h1');
// Will throw an error because more than 1 matches were found...
const result2 = findVNodeWithType(VNodeTree, SomeComponent);
```

### `isVNode(instance)`

Returns `true` when `instance` is a `VNode`.

### `isVNodeOfType(instance, type)`

Returns `true` when `instance` is a `VNode` of `type`.

### `isDOMVNode(instance)`

Returns `true` when `instance` is a DOM-type `VNode`.

### `isDOMVNodeOfType(instance, type)`

Returns `true` when `instance` is a DOM-type `VNode` of `type`.

### `isFunctionalVNode(instance)`

Returns `true` when `instance` is a functional-type `VNode`.

### `isFunctionalVNodeOfType(instance, type)`

Returns `true` when `instance` is a functional-type `VNode` of `type`.

### `isClassVNode(instance)`

Returns `true` when `instance` is a class-type `VNode`.

### `isClassVNodeOfType(instance, type)`

Returns `true` when `instance` is a class-type `VNode` of `type`.

### `isDOMElement(instance)`

Returns `true` when `instance` is a DOM element.

### `isDOMElementOfType(instance, type)`

Returns `true` when `instance` is a DOM element of `type`.

### `isRenderedClassComponent(instance)`

Returns `true` when `instance` is a rendered class `VNode`.

### `isRenderedClassComponentOfType(instance, type)`

Returns `true` when `instance` is a rendered class `VNode` of `type`.
