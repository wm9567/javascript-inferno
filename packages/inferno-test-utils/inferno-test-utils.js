module.exports = require('inferno/dist/inferno-test-utils.node');
module.exports.default = module.exports;