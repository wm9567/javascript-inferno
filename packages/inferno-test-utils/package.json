{
  "name": "inferno-test-utils",
  "version": "0.8-alpha",
  "license": "MIT",
  "description": "Provides test utilities to help with creating and maintaining tests for Inferno applications",
  "author": {
    "name": "Dominic Gannaway",
    "email": "dg@domgan.com"
  },
  "bugs": {
    "url": "https://github.com/trueadm/inferno/issues"
  },
  "homepage": "https://github.com/trueadm/inferno#readme",
  "keywords": [
    "babel",
    "react",
    "inferno",
    "framework",
    "interfaces",
    "user interfaces"
  ],
  "dependencies": {
    "inferno": "0.8-alpha"
  },
  "main": "inferno-test-utils.js",
  "repository": "https://github.com/trueadm/inferno"
}
