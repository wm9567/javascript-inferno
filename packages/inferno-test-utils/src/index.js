import {
	findAllInRenderedTree,
	findRenderedDOMComponentWithClass,
	isCompositeComponent,
	isCompositeComponentWithType,
	isDOMComponent,
	isDOMComponentElement,
	isElement,
	isElementOfType,
	renderIntoDocument,
	scryRenderedDOMComponentsWithClass,
	scryRenderedDOMComponentsWithTag,
	scryRenderedComponentsWithType,
	findRenderedDOMComponentWithTag,
	findRenderedComponentWithType,
	mockComponent
} from '../../../build/tools/testUtils';

export {
	renderIntoDocument,
	isElement,
	isElementOfType,
	isDOMComponent,
	isDOMComponentElement,
	isCompositeComponent,
	isCompositeComponentWithType,
	findAllInRenderedTree,
	scryRenderedDOMComponentsWithClass,
	scryRenderedDOMComponentsWithTag,
	scryRenderedComponentsWithType,
	findRenderedDOMComponentWithClass,
	findRenderedDOMComponentWithTag,
	findRenderedComponentWithType,
	mockComponent
};

export default {
	renderIntoDocument,
	isElement,
	isElementOfType,
	isDOMComponent,
	isDOMComponentElement,
	isCompositeComponent,
	isCompositeComponentWithType,
	findAllInRenderedTree,
	scryRenderedDOMComponentsWithClass,
	scryRenderedDOMComponentsWithTag,
	scryRenderedComponentsWithType,
	findRenderedDOMComponentWithClass,
	findRenderedDOMComponentWithTag,
	findRenderedComponentWithType,
	mockComponent,
};
