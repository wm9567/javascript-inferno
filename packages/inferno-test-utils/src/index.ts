import { InfernoChildren, InfernoInput, render, VNode } from 'inferno';
import Component from 'inferno-component';
import createElement from 'inferno-create-element';
import { isArray, isFunction, isNumber, isObject, isString, throwError } from 'inferno-shared';
import VNodeFlags from 'inferno-vnode-flags';

// Type Checkers

export function isVNode(instance: any): boolean {
	return Boolean(instance) && isObject(instance) &&
		isNumber(instance.flags) && instance.flags > 0;
}

export function isVNodeOfType(instance: VNode, type: string | Function): boolean {
	return isVNode(instance) && instance.type === type;
}

export function isDOMVNode(instance: VNode): boolean {
	return isVNode(instance) && isString(instance.type);
}

export function isDOMVNodeOfType(instance: VNode, type: string): boolean {
	return isDOMVNode(instance) && instance.type === type;
}

export function isFunctionalVNode(instance: VNode): boolean {
	return isVNode(instance) && Boolean(instance.flags & VNodeFlags.ComponentFunction);
}

export function isFunctionalVNodeOfType(instance: VNode, type: Function): boolean {
	return isFunctionalVNode(instance) && instance.type === type;
}

export function isClassVNode(instance: VNode): boolean {
	return isVNode(instance) && Boolean(instance.flags & VNodeFlags.ComponentClass);
}

export function isClassVNodeOfType(instance: VNode, type: Function): boolean {
	return isClassVNode(instance) && instance.type === type;
}

export function isDOMElement(instance: any): boolean {
	return Boolean(instance) && isObject(instance) &&
		instance.nodeType === 1 && isString(instance.tagName);
}

export function isDOMElementOfType(instance: any, type: string): boolean {
	return isDOMElement(instance) && isString(type) &&
		instance.tagName.toLowerCase() === type.toLowerCase();
}

export function isRenderedClassComponent(instance: any): boolean {
	return Boolean(instance) && isObject(instance) && isVNode(instance._vNode) &&
		isFunction(instance.render) && isFunction(instance.setState);
}

export function isRenderedClassComponentOfType(instance: any, type: Function): boolean {
	return isRenderedClassComponent(instance) &&
		isFunction(type) && instance._vNode.type === type;
}

// Render Utilities

class Wrapper extends Component<any, any> {
	render() {
		return this.props.children;
	}
}

export function renderIntoDocument(input: InfernoInput): InfernoChildren {
	const wrappedInput = createElement(Wrapper, null, input);
	const parent = document.createElement('div');
	document.body.appendChild(parent);
	return render(wrappedInput, parent);
}

// Recursive Finder Functions

export function findAllInRenderedTree(renderedTree: any, predicate: (vNode: VNode) => boolean): VNode[] {
	if (isRenderedClassComponent(renderedTree)) {
		return findAllInVNodeTree(renderedTree._lastInput, predicate);
	} else {
		throwError('findAllInRenderedTree(renderedTree, predicate) renderedTree must be a rendered class component');
	}
}

export function findAllInVNodeTree(vNodeTree: VNode, predicate: (vNode: VNode) => boolean): VNode[] {
	if (isVNode(vNodeTree)) {
		let result: VNode[] = predicate(vNodeTree) ? [ vNodeTree ] : [];
		const children: any = vNodeTree.children;

		if (isRenderedClassComponent(children)) {
			result = result.concat(findAllInVNodeTree(children._lastInput, predicate));

		} else if (isVNode(children)) {
			result = result.concat(findAllInVNodeTree(children, predicate));

		} else if (isArray(children)) {
			children.forEach((child) => {
				result = result.concat(findAllInVNodeTree(child, predicate));
			});
		}
		return result;
	} else {
		throwError('findAllInVNodeTree(vNodeTree, predicate) vNodeTree must be a VNode instance');
	}
}

// Finder Helpers

function parseSelector(filter) {
	if (isArray(filter)) {
		return filter;
	} else if (isString(filter)) {
		return filter.trim().split(/\s+/);
	} else {
		return [];
	}
}

function findOneOf(tree: any, filter: any, name: string, finder: Function): any {
	const all = finder(tree, filter);
	if (all.length > 1) {
		throwError(`Did not find exactly one match (found ${all.length}) for ${name}: ${filter}`);
	} else {
		return all[ 0 ];
	}
}

// Scry Utilities

export function scryRenderedDOMElementsWithClass(renderedTree: any, classNames: string | string[]): Element[] {
	return findAllInRenderedTree(renderedTree, (instance) => {
		if (isDOMVNode(instance)) {
			let domClassName = instance.dom.className;
			if (!isString(domClassName)) { // SVG, probably
				domClassName = instance.dom.getAttribute('class') || '';
			}
			const domClassList = parseSelector(domClassName);
			return parseSelector(classNames).every((className) => {
				return domClassList.indexOf(className) !== -1;
			});
		}
		return false;
	}).map((instance) => instance.dom);
}

export function scryRenderedDOMElementsWithTag(renderedTree: any, tagName: string): Element[] {
	return findAllInRenderedTree(renderedTree, (instance) => {
		return isDOMVNodeOfType(instance, tagName);
	}).map((instance) => instance.dom);
}

export function scryRenderedVNodesWithType(renderedTree: any, type: string | Function): VNode[] {
	return findAllInRenderedTree(renderedTree, (instance) => isVNodeOfType(instance, type));
}

export function scryVNodesWithType(vNodeTree: VNode, type: string | Function): VNode[] {
	return findAllInVNodeTree(vNodeTree, (instance) => isVNodeOfType(instance, type));
}

// Find Utilities

export function findRenderedDOMElementWithClass(renderedTree: any, classNames: string | string[]): Element {
	return findOneOf(renderedTree, classNames, 'class', scryRenderedDOMElementsWithClass);
}

export function findRenderedDOMElementWithTag(renderedTree: any, tagName: string): Element {
	return findOneOf(renderedTree, tagName, 'tag', scryRenderedDOMElementsWithTag);
}

export function findRenderedVNodeWithType(renderedTree: any, type: string | Function): VNode {
	return findOneOf(renderedTree, type, 'component', scryRenderedVNodesWithType);
}

export function findVNodeWithType(vNodeTree: VNode, type: string | Function): VNode {
	return findOneOf(vNodeTree, type, 'VNode', scryVNodesWithType);
}

export default {

	isVNode,
	isVNodeOfType,

	isDOMVNode,
	isDOMVNodeOfType,

	isFunctionalVNode,
	isFunctionalVNodeOfType,

	isClassVNode,
	isClassVNodeOfType,

	isDOMElement,
	isDOMElementOfType,

	isRenderedClassComponent,
	isRenderedClassComponentOfType,

	renderIntoDocument,

	findAllInRenderedTree,
	findAllInVNodeTree,

	scryRenderedDOMElementsWithClass,
	findRenderedDOMElementWithClass,

	scryRenderedDOMElementsWithTag,
	findRenderedDOMElementWithTag,

	scryRenderedVNodesWithType,
	findRenderedVNodeWithType,

	scryVNodesWithType,
	findVNodeWithType
};
