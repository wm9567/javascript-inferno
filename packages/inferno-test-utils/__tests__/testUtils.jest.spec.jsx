import { createElement } from 'inferno-create-element';
import * as TestUtils from 'inferno-test-utils';

const FunctionalComponent = function(props) {
  return createElement('div', props);
};

describe('renderToSnapshot', () => {
  it('should return a snapshot from a valid vNode', () => {
    const snapshot = TestUtils.renderToSnapshot(
      <FunctionalComponent foo="bar" />
    );

    if (usingJest) {
      expect(snapshot).toMatchSnapshot();
    } else {
      expect(snapshot.props.foo).toBeDefined();
      expect(snapshot.props.foo).toBe('bar');
    }
  });

  it('should return a snapshot with className prop', () => {
    const TestComponent = () => <div className="test">Test</div>;

    const snapshot = TestUtils.renderToSnapshot(<TestComponent />);

    if (usingJest) {
      expect(snapshot).toMatchSnapshot();
    } else {
      expect(snapshot.props.className).toBe('test');
    }
  });

  it('should return a snapshot with className prop, multiple children', () => {
    const TestComponent = (props) => <div className="test">{props.children}<span>1</span></div>;

    const snapshot = TestUtils.renderToSnapshot(<TestComponent>
      {[
        <span>a</span>,
        <span>b</span>
      ]}
    </TestComponent>);

    if (usingJest) {
      expect(snapshot).toMatchSnapshot();
    } else {
      expect(snapshot.props.className).toBe('test');
    }
  });
});
