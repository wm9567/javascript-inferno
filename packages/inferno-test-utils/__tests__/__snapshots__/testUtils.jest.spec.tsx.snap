// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderToSnapshot should return a snapshot from a valid vNode 1`] = `
<div
  foo="bar"
/>
`;

exports[`renderToSnapshot should return a snapshot with className prop 1`] = `
<div
  className="test"
>
  Test
</div>
`;

exports[`renderToSnapshot should return a snapshot with className prop, multiple children 1`] = `
<div
  className="test"
>
  <span>
    a
  </span>
  <span>
    b
  </span>
  <span>
    1
  </span>
</div>
`;
